src/prefect/agent/vertex/agent.py

```diff
@@ -0,0 +1,279 @@
+        }
+        self.vertex_client = aiplatform.gapic.JobServiceClient(
+            client_options=self.client_options
+        )
+
+    def deploy_flow(self, flow_run: GraphQLResult) -> str:
+        """
+        Deploy a flow run as an Vertex task.
+
+        Args:
+            - flow_run (GraphQLResult): A GraphQLResult flow run object
+
+        Returns:
+            - str: Information about the deployment
+        """
+        run_config = self._get_run_config(flow_run, VertexRun)
+        assert isinstance(run_config, VertexRun)  # mypy
+        taskdef = self.generate_task_definition(flow_run, run_config)
+        parent = f"projects/{self.project}/locations/{self.region_name}"
+
+        self.logger.debug(
+            "Created task definition for flow %s",
+            flow_run.flow.id,
+        )
+
+        resp = self.vertex_client.create_custom_job(parent=parent, custom_job=taskdef)
+
+    def generate_task_definition(
+        self, flow_run: GraphQLResult, run_config: VertexRun
+    ) -> Dict[str, Any]:
+        """Generate an Vertex task definition from a flow run
+
+        Args:
+            - flow_run (GraphQLResult): A flow run object
+            - run_config (VertexRun): The flow's run config
+
+        Returns:
+            - dict: a dictionary representation of an Vertex task definition
+        """
+
+        slug = slugify.slugify(
+            flow_run.flow.name,
+            max_length=255 - len("prefect-"),
+            word_boundary=True,
+            save_order=True,
+        )
+
+        # Set flow image
+        run_config.image = None
+        image = get_flow_image(flow_run, default=run_config.image)
+
+        if hasattr(run_config, "display_name"):
+            job_name = run_config.display_name
+        else:
+            job_name = f"prefect-{slug}"
+
+        machine_type = run_config.machine_type
+
+        boot_disk_size_gb = run_config.boot_disk_size_gb
+
+        command = get_flow_run_command(flow_run)
+        env = self.populate_env_vars(flow_run, image)
+        env_list = []
+        for k, v in env.items():
+            env_list.append({"name": k, "value": v})
+        taskdef = {
+            "display_name": job_name,
+            "job_spec": {
+                "worker_pool_specs": [
+                    {
+                        "machine_spec": {"machine_type": machine_type},
+                        "replica_count": 1,
+                        "disk_spec": {
+                            "boot_disk_type": "pd-ssd",
+                            "boot_disk_size_gb": int(boot_disk_size_gb),
+                        },
+                        "container_spec": {
+                            "image_uri": image,
+                            "command": command.split(),
+                            "args": [],
+                            "env": env_list,
+                        },
+                    }
+                ]
+            },
+        }
+
+        if hasattr(run_config, "network"):
```

Comment:
now that we have our own run config and not the generic one, we should probably not hasattr here but instead rely on it having a default, so like `if run_config.network is not None`

Reply:
Yeah, just missed that one.